### PR TITLE
[posix] fix udp close

### PR DESCRIPTION
**Pull request by librasungirl**
Fixes https://github.com/openthread/ot-br-posix/issues/561.

### CODE REVIEW

**Review by bukepo — APPROVED** [review]
LGTM:+1:

**Review by bukepo — APPROVED** [review]

**Review by suveshpratapa — APPROVED** [review]
LGTM

File: src/posix/platform/udp.cpp [hunk 1]
```diff
@@ -240,7 +240,10 @@ otError otPlatUdpClose(otUdpSocket *aUdpSocket)
     otError error = OT_ERROR_NONE;
     int     fd;
 
-    VerifyOrExit(aUdpSocket->mHandle != nullptr, error = OT_ERROR_INVALID_ARGS);
+    // Here skips further close ops for non Platform UDP socket.
+    // Platform UDP always has valid mHandle when create.
+    VerifyOrExit(aUdpSocket->mHandle != nullptr);
```
Comment (simonlingoogle) [inline src/posix/platform/udp.cpp:245]: Whether it's platform UDP or not, returning success should be the best choice here.

**Review by simonlingoogle — APPROVED** [review]
LGTM 👍

**Review by jwhui — APPROVED** [review]

**Comment by jwhui** [comment]
@librasungirl , is this still work-in-progress?

File: src/posix/platform/udp.cpp [hunk 2]
```diff
@@ -240,7 +240,10 @@ otError otPlatUdpClose(otUdpSocket *aUdpSocket)
     otError error = OT_ERROR_NONE;
     int     fd;
 
-    VerifyOrExit(aUdpSocket->mHandle != nullptr, error = OT_ERROR_INVALID_ARGS);
+    // Here skips further close ops for non Platform UDP socket.
+    // Platform UDP always has valid mHandle when create.
```
Comment (jwhui) [inline src/posix/platform/udp.cpp:244]:
```suggestion
    // Only call `close()` on platform UDP sockets.
    // Platform UDP sockets always have valid `mHandle` upon creation.
```

**Pull request merged**